docs/content/basics/ui_testing.md

```diff
@@ -1,4 +1,119 @@
 # UI testing
 
-To be done
+UI tests is a part of instrumentation tests.
+That's why everything from <link> Instrumented tests topic everything is applicable to UI testing.
 
+## Main tool - Espresso
+
+Nowadays there is no doubt - for native UI testing on Android we should use Espresso for testing your application.
+<br/> This is a main tool that allow us to make every test possible to access your codebase. That means tests with Espresso allow us to write white-box tests.
+<br/>But you still able to write black-box tests or even gray-box with Espresso.
+
+Why Espresso is so cool? It is synchronized with the main thread of your app and performs action only when main thread is in idle state.
+<br/>It makes Espresso extremely reliable and stable (than any other existing tool) for Ui testing on Android.
+
+Keep in mind that everything that isn't related to your app (for example, permission dialogs, volume +/- buttons, home button and etc) - can't be accessed directly within Espresso.
+<br/>But don't worry - you can use `UiAutomator` for that purpose, and it can be called directly from test, written in Espresso.
+<br/>`UiAutomator` is a slower tool and can support tests only in black-box style, but it can test everything outside your application on your phone.
+<br/>Many cross-platforms tools for testing using `UiAutomator`.
+
+### Where to start
+
+Firstly, you should add dependency for Espresso library to your Android app module
+
+```kotlin
+dependencies {
+    androidTestImplementation('androidx.test.espresso:espresso-core:3.4.0')
```

Comment:
I'm not really sure that it's worth to have a documentation description here. We can just point to the android documentation

Reply:
allright, let's delete that